Start a session D-Bus before anything else.
Then register ourselves with ConsoleKit via $STARTUP

--- data/scripts/Xsession.orig	2023-06-23 12:28:38 UTC
+++ data/scripts/Xsession
@@ -90,5 +90,5 @@ else
 if [ -z "$*" ]; then
     exec xmessage -center -buttons OK:0 -default OK "Sorry, $DESKTOP_SESSION is no valid session."
 else
-    exec $@
+    exec %%LOCALBASE%%/bin/dbus-launch --exit-with-x11 -- $STARTUP "$@"
 fi
